### ISSUE TITLE
Listen on 127.0.0.1 rather than 0.0.0.0

### ISSUE THREAD

**Issue by alecthomas** [livebud/bud#35]
OSX throws up a security prompt every time bud starts because it's binding to 0.0.0.0. Binding to 127.0.0.1 instead will rectify this.

![image](https://user-images.githubusercontent.com/41767/168423363-3d44e595-40e5-4e4f-b2bb-bc8ca55b5474.png)

**Comment by alecthomas** [comment]
Ah I see by digging into the code that this is actually possible to override:

```
bud run --port 127.0.0.1:3000
```

I didn't expected that to work because the flag is named `--port`, rather than eg. `--bind`.

**Comment by syke99** [comment]
@alecthomas not yet a contributor (though I plan to once my other projects calm down just a bit) but how do y’all feel about removing the need to pass in a `—port flag`, adding in a check `run` to determine to user’s OS and bind to 0.0.0.0 if anything but OSX?

**Comment by alecthomas** [comment]
IMO it's fine to bind development servers to 127.0.0.1 by default. As long as there's a way to change it, there's no real downside.

**Comment by syke99** [comment]
I don’t either, but was unaware of the team’s view and wanted to toss out an option for if defaulting to 127.0.0.1 was shot down. Thanks!!

**Comment by amandamcg** [comment]
> OSX throws up a security prompt every time bud starts because it's binding to 0.0.0.0. Binding to 127.0.0.1 instead will rectify this.
> 
> ![image](https://user-images.githubusercontent.com/41767/168423363-3d44e595-40e5-4e4f-b2bb-bc8ca55b5474.png)

On my machine I didn't even get a warning, it just wouldn't load. Loading as `http://localhost:3000/` worked fine in case it helps anyone else.

**Comment by quaintdev** [comment]
I agree we should default to 127.0.0.1. But I think while building for production it should default to 0.0.0.0. I am not sure if bud is aware of dev/prod environments though. I guess passing the IP seems easiest solution for now. 

I don't like the fact that `--port 127.0.0.1:3000` can change binding address too. Probably we should change '--port` to `--bind`.

**Comment by syke99** [comment]
Is there a reason that there’s not separate functionality for changing the binding address and port separately with `—bind` and `—port`, respectively? Seems doing so could possibly open up a solution for determining dev vs. prod. Especially in the context of CI/CD

**Comment by codenoid** [comment]
I will try to work on this, --host and --port separation

**Comment by matthewmueller** [comment]
PR to change the default to `127.0.0.1`. 

I've created [a discussion](https://github.com/livebud/bud/discussions/42) for changing `--port` to something better. Please share your thoughts there!

**Comment by alecthomas** [comment]
🙏

**Comment by matthewmueller** [comment]
FYI. This is available in [v0.1.1](https://github.com/livebud/bud/releases/tag/v0.1.1). To upgrade, run:

```sh
curl -sf curl https://raw.githubusercontent.com/livebud/bud/main/install.sh | sh
```
